### PR TITLE
[Bug 17239] json: Add tests for number formatting/parsing

### PR DESCRIPTION
**Pull request by peter-b**
- Add failing test for lossless encoding of integers
- Add passing test for parsing of numbers in scientific format

### CODE REVIEW

File: extensions/libraries/json/tests/numbers.livecodescript [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+on TestExportInteger
+	local tArray, tExpected
+	put (14592610+0) into tArray[1]
+	put "{" & quote & "1" & quote & ": 14592610}" into tExpected
+
+	local tJson
+	put JsonExport(tArray) into tJson
+	TestDiagnostic tJson
+	TestDiagnostic tExpected
+	TestAssertBroken "export large integer losslessly", tExpected is tArray, \
```
Comment (livecodeali) [inline extensions/libraries/json/tests/numbers.livecodescript]: I think this should be `tExpected is tJson`

**Comment by peter-b** [comment]
@livecodeali Pull request updated! Thanks!

**Comment by livecodeali** [comment]
@livecode-vulcan review ok 39a7603

**Comment by livecode-vulcan** [comment]
💙 review by @livecodeali ok 39a7603fa5eb250932c35de425c2d2103928fb9c

**Comment by livecode-vulcan** [comment]
😎 test success 39a7603fa5eb250932c35de425c2d2103928fb9c
- try-community-linux-x86: [success](https://vulcan.livecode.com/builders/try-community-linux-x86/builds/2244)
- try-community-mac: [success](https://vulcan.livecode.com/builders/try-community-mac/builds/2252)
- try-community-linux-x86_64: [success](https://vulcan.livecode.com/builders/try-community-linux-x86_64/builds/2237)
- try-community-emscripten: [success](https://vulcan.livecode.com/builders/try-community-emscripten/builds/2253)
- try-community-ios: [success](https://vulcan.livecode.com/builders/try-community-ios/builds/2259)
- try-community-win-x86: [success](https://vulcan.livecode.com/builders/try-community-win-x86/builds/2265)
- try-community-android-armv6: [success](https://vulcan.livecode.com/builders/try-community-android-armv6/builds/2275)

**Pull request merged**